File: test/rs/case15_class_member.rs
#[path = "../../src/rs_module/py_runtime.rs"]
mod py_runtime;
use py_runtime::{perf_counter, py_in, py_len, py_print, py_slice};

// このファイルは自動生成です（native Rust mode）。

struct Counter {
    value: i64,
}

impl Counter {
    fn new() -> Self {
        let mut self_obj = Self {
            value: 0,
        };
        self_obj
    }

    fn inc(&mut self) -> i64 {
        self.value = self.value + 1;
        return self.value;
    }
}

fn main() {
    let mut c: Counter = Counter::new();
    py_print(c.inc());
}

File: test/cpp/case15_class_member.cpp
// このファイルは `test/py/case15_class_member.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。


#include "cpp_module/py_runtime.h"

// このファイルは `test/py/case15_class_member.py` のテスト/実装コードです。
// 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
// 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。

struct Counter {
    inline static int64 value = 0;
    
    int64 inc() {
        Counter::value++;
        return Counter::value;
    }
};

int main() {
    Counter c = Counter();
    c.inc();
    c = Counter();
    py_print(c.inc());
    return 0;
}
